Automattic/sensei
# Enable to use page template on the new emails

Resolves #6575 

Please notice, the PR #6677 fixed layout issues, so in this PR ignore the layout issue on this PR

## Proposed Changes
* Enable be possible to use of the regular page templates on the new emails
* Ensure compatibility ONLY with the Welcome email, the other emails will be updated late. 
* Set all email templates to use the `Sensei Email` template as the default template. 

## Testing instructions
* Enable the feature toggle
* Run npm run start
* Run the `Email` creation tool
* Go to an e-mail editor
* Click on Edit template
* Edit the template
* Go to the preview
* Check if the change was applied
* Assign a student to a course
* Check the Welcome email is showing the correct header and footer

<img width="1293" alt="Screenshot 2023-03-17 at 18 06 57" src="https://user-images.githubusercontent.com/38718/225984356-40e0f95f-4389-4a0a-974c-00ea01be014d.png">


## Extra notes
* ~The email preview seems a bit unstable, so a new card will be created to try to fix it.~ Fixed
* If the user selects any other template to render the emails, it will be ignored in order to use the email template.


## Technical Contexts
WordPress templates are very related to themes, so every time WP tries to look for a template revision, it tries to find one using the theme. In your case, we don't create a new theme, so we need to help WP to find ou new template overriding some hooks, using the following sequence: 

The process to enable the usage of templates is a trick and requires multiple hooks working together. 

1) This page calls the `get_templates`, [code here](https://github.com/Automattic/sensei/blob/1eda3525b1c9570a5f71bcc8c117644fee1f9c94/includes/internal/emails/class-email-page-template.php#L60)
<img width="1374" alt="Screenshot 2023-03-17 at 17 54 51" src="https://user-images.githubusercontent.com/38718/225981769-6f1671e1-3f94-47e8-aa04-d67344f7bbc8.png">

At this moment, is required to introduce the new template + its content, from the file.

2) When the user clicks on the template name, no new hook is called because WordPress cached the value from the previous hook.
<img width="2258" alt="Screenshot 2023-03-17 at 17 57 43" src="https://user-images.githubusercontent.com/38718/225982376-e58f70c8-b00f-47c4-8045-b14a20c34e4a.png">

But, if you reload this page, WordPress  tries to find the template on the database and when it fails, the hook `pre_get_block_file_template` is called, code [here](https://github.com/Automattic/sensei/blob/1eda3525b1c9570a5f71bcc8c117644fee1f9c94/includes/internal/emails/class-email-page-template.php#L59)

This hook is also called when clicking on the `Revert to the original` version button.


3) When we need to access the template during the rendering process, WP calls the `get_template` ([code here](https://github.com/Automattic/sensei/blob/1eda3525b1c9570a5f71bcc8c117644fee1f9c94/includes/internal/emails/class-email-page-template.php#L62))


## Pre-Merge Checklist
